lib/shared/bucketing-assembly-script/assembly/types/target.ts

```diff
@@ -43,65 +43,112 @@ export class Target extends JSON.Value {
     }
 }
 
-export class NoIdAudience extends JSON.Value {
-    filters: TopLevelOperator
+export class AudienceFilter extends JSON.Value {
+    type: string
+    constructor(filter: JSON.Obj) {
+        super()
+        this.type = isValidString(filter, 'type', validTypes, false)
 
-    constructor(audience: JSON.Obj) {
+    }
+
+    stringify(): string {
+        const json = new JSON.Obj()
+        if (this.type) {
+            json.set('type', this.type)
+        }
+
+        return json.stringify()
+    }
+}
+// This is a virtual container class that's used by the code to determine whether a filter is a nested operator
+// or a base level filter. This is used to support the recursive nature of nested filters.
+// The config is returned to its original form when stringified.
+export class AudienceFilterOrOperator extends JSON.Value {
+    operatorClass: AudienceOperator | null
+    filterClass: AudienceFilter | null
+
+    constructor(filter: JSON.Obj) {
+        super()
+        const operator = isValidStringOptional(filter, 'operator', validAudienceOperators)
```

Comment:
this should be passing false to "shouldThrow".

might need a test to cover this

Reply:
it actually shouldnt because its using `isValidStringOptional`, so it wont throw

Reply:
afaik the Optional version only doesnt throw when the field is actually null, if it's set to a value that isnt in the set of valid types it throws